### DISCUSSION TITLE
Is it possible to use complex JSONPath expression with dynamic badge? Such as string concatenating, ternary expression.

### DISCUSSION THREAD

**Discussion by razonyang** [badges/shields#10468] [General] [answered]
For example,

```json
{
  "min": "0.1.0"
  "max": "1.0.0"
  "ext": true
}
```

I wish to generate badge like `requirements | 0.1.0~1.0.0 extended`.

Is there something like `$.concat($.min, "~", $.max, $.ext == true ? 'ext' : '')`.

**Comment by chris48s** [comment, marked as answer]
I haven't tried your example, but I suspect the answer is no.

The library we use to evaluate JSONPath expressions is https://www.npmjs.com/package/jsonpath there's some info in the README about the syntax that is compatible. Note there is quite a lot of subtle variation between JSONPath libraries. https://cburgmer.github.io/json-path-comparison/ is the canonical resource for comparing them.

Is that example you posted (with concat and ternary) synthx that is compatible with some JSONPath libraries, or just syntax you wish existed? I had a quick scan of the RFC https://www.rfc-editor.org/rfc/rfc9535 and I'm not convinced that is valid for any implementation.

**Reply by razonyang** [reply]
Thanks for the detail, I may need to request features from upstream library.

The ternary expression is sth I want, maybe not valid for any implementation, and the `concat` function is valid for some implementation.

![image](https://github.com/user-attachments/assets/124d3b0a-90f9-4cb4-9199-4cd9004635ec)

> Captured from https://www.site24x7.com/tools/json-path-evaluator.html.

**Reply by chris48s** [reply]
TIL :mortar_board:
